components/style/viewport.rs

```diff
@@ -685,6 +685,7 @@ impl MaybeNew for ViewportConstraints {
         let initial_viewport = device.au_viewport_size();
 
         // TODO(emilio): Stop cloning `ComputedValues` around!
+        #[cfg(not(feature = "gecko"))]
         let context = Context {
```

Comment:
Perhaps `Context` should have a constructor that takes a `Device`? This seems quite inconvenient :(.

Also, perhaps it should hold a `&'a Device` unconditionally, and just access the prescontext from Gecko. We could then get rid of the `viewport_size` variable, and access just `device.viewport_size`, what do you think?